src/Events/Calendar_Embeds/Calendar_Embeds.php

```diff
@@ -115,22 +231,87 @@ public function register_post_type() {
 		 */
 		$args = apply_filters( 'tec_events_calendar_embeds_post_type_args', $args );
 
-		$this->post_type_object = register_post_type( static::POSTTYPE, $args );
+		register_post_type( static::POSTTYPE, $args );
 	}
 
 	/**
-	 * Get the post type object.
+	 * Get the iframe code for the calendar embed.
 	 *
 	 * @since TBD
 	 *
-	 * @return \WP_Post_Type
-	 * @throws \RuntimeException If the post type object is not set.
+	 * @param int $post_id The post ID.
+	 *
+	 * @return string
+	 * @throws NotPublishedCalendarException When the calendar is not published.
 	 */
-	public function get_post_type_object() {
-		if ( ! $this->post_type_object ) {
-			throw new \RuntimeException( __( 'Attempted to get post type object before it was set.', 'the-events-calendar' ) );
+	public static function get_iframe( int $post_id, bool $throw_when_not_published = false ): string {
+		$embed = get_post( $post_id );
+
+		if ( ! $embed instanceof WP_Post ) {
+			return '';
+		}
+
+		if ( static::POSTTYPE !== $embed->post_type ) {
+			return '';
+		}
+
+		if ( $throw_when_not_published && 'publish' !== $embed->post_status ) {
+			throw new NotPublishedCalendarException();
+		}
+
+		$embed_url = 'publish' === $embed->post_status ? get_post_embed_url( $embed ) : get_preview_post_link( $embed, [ 'embed' => 1 ] );
+
+		$iframe = '<iframe src="' . esc_url( $embed_url ) . '" width="100%" height="600" style="max-width:100%;" frameborder="0"></iframe>';
+
+		/**
+		 * Filter the iframe code for the calendar embed.
+		 *
+		 * @since TBD
+		 *
+		 * @param string  $iframe    The iframe code.
+		 * @param WP_Post $embed     The embed post object.
+		 * @param string  $embed_url The embed URL.
+		 *
+		 * @return string
+		 */
+		return (string) apply_filters( 'tec_events_calendar_embeds_iframe', $iframe, $embed, $embed_url );
```

Comment:
This filter gives users the power to modify the iframe HTML - which could lead to all kinds of mischief (security). The only use I see is for the draft preview functionality, maybe it shouldn't be a filter?

If we want to let folks alter the props (width, height, style), we can filter those but (for example) do we want them to alter `src`?

Reply:
I would not be against to filtering the iframe attrs.

But i am also not against of filtering the whole iframe markup. My opinion is that if you are writting code for your site you should know what you are doing. So if you make your site insecure because of code you wrote, thats on you :(

if you dont know what you are doing, dont write code for your site.

Reply:
Also if you get a vulnerability by someone else adding malicious code into your site then you have a more serious problem than what this filter allows already :D